### ISSUE TITLE
when leaving the domain the label doesn't change right away

### ISSUE THREAD

**Issue by andreasn** [cockpit-project/cockpit#11631]
Discovered this when doing usability testing last week.

Cockpit version: 191
OS: Fedora
Page: System

* Join a domain
* Leave the domain
* The dialog will process, and close at some point, but the address won't change back to `Join Domain` right away, making it unclear if the machine is still joined to the domain or not.

**Comment by martinpitt** [comment]
Confirmed on current cockpit.

**Comment by martinpitt** [comment]
I had a first look at this -- [realm.Deconfigure()](https://freedesktop.org/software/realmd/docs/gdbus-org.freedesktop.realmd.Realm.html) is asynchronous, and there is no flag for waiting for the leaving to actually finish. We would have to add a hack like "keep the dialog open until we get a `changed` signal.

**Comment by marusak** [comment]
I tried to reproduce this and was not able nor on fedora-36 nor rhel-8-6. The dialog stays open and closes only once the text is changed to `Join a domain`.

So I looked if there were any changes lately in the related code and I found https://github.com/cockpit-project/cockpit/pull/16429

So I did this tiny change: (#16429 changed this the other way)
```
--- pkg/systemd/overview-cards/realmd.jsx
+++ pkg/systemd/overview-cards/realmd.jsx
@@ -155,7 +155,7 @@ export class RealmdClient {
 
     leave(realm) {
         return this.cleanupWSCredentials(realm)
-                .then(() => realm.Deconfigure({ operation: cockpit.variant('s', "cockpit-leave-domain") }));
+                .then(function() { realm.Deconfigure({ operation: cockpit.variant('s', "cockpit-leave-domain") }) });
     }
```
and indeed, now I can reproduce it just fine. The trick is, that when using `{}` the default return is `undefined` but when using this syntax without `{}` it returns the `Deconfigure` promise. Changing it to `...function() { return realm.Deconfigure(...` makes it work again.

So one more win for rewriting it to more modern JS :tada:
